hw_2/hw_2.py

```diff
@@ -0,0 +1,38 @@
+import random
+import string
+
+
+def generate_random_dict():
+    # Random number of keys
+    num_keys = random.randint(2, 3)
+    # Random letters as keys
+    keys = random.choices(string.ascii_lowercase, k=num_keys)
+    values = [random.randint(0, 100) for _ in range(num_keys)]
+    # Combine keys and values into a dictionary
+    return dict(zip(keys, values))
+
+
+def generate_list_of_random_dicts():
+    num_dicts = random.randint(2, 10)
+    # Random number of dictionaries in the list
+    return [generate_random_dict() for _ in range(num_dicts)]
+
+
+def common_dict(list_of_dicts):
+    merged_dict = {}
+    for i, dictionary in enumerate(list_of_dicts, start=1):
+        for key, value in dictionary.items():
+            if key in merged_dict:
```

Comment:
default key is compared with {key}_{i} format:

[{'a':12, 'b':20}, {'b':30},{'b':5}]